### PR TITLE
logic added to check claims status on deposit and revert if not ready…

### PR DESCRIPTION
**Pull request by stephenbyrne99**
…, tests working for ready/investigating status but NOT paid status.

### CODE REVIEW

File: contracts/claimsManagers/ClaimsManagerSingleAccount.sol [hunk 1]
```diff
@@ -25,6 +25,7 @@ contract ClaimsManagerSingleAccount {
     	currentInvestigationPeriodEnd = 0;
     	activePayoutEvent = false;
     	governance = msg.sender;
+		status = ClaimsStatus.Ready;
```
Comment (corbinpage) [inline contracts/claimsManagers/ClaimsManagerSingleAccount.sol:28]: @stephenbyrne99 can you fix the indentation here?

File: contracts/protektCore/pToken.sol [hunk 2]
```diff
@@ -7,6 +7,8 @@ import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
 import "@openzeppelin/contracts/token/ERC20/ERC20Detailed.sol";
 import "./helpers/HarvestRewardsCompoundDaiManual.sol";
+import "../claimsManagers/interfaces/IClaimsManagerCore.sol";
+import "../claimsManagers/ClaimsManagerSingleAccount.sol";
```
Comment (corbinpage) [inline contracts/protektCore/pToken.sol:11]: @stephenbyrne99 I don't think you need to import this contract. I believe the interface should be enough. See what you think.

File: migrations/2_deploy_new_protekt_contract.js [hunk 3]
```diff
@@ -20,14 +20,22 @@ module.exports = function (deployer, network, accounts) {
...
-  // console.log('accounts')
-  // console.log(accounts)
-  // console.log('---')
-  // console.log(accounts[0])
+
+    /*
+        Move claims manager before pToken so could use it when deploying
+    */
+
+    // 4) Launch ClaimsManager (ClaimsManagerSingleAccount) ==============
```
Comment (corbinpage) [inline migrations/2_deploy_new_protekt_contract.js:28]: @stephenbyrne99 please just reorder the comments if you could.

File: tests/pToken.js [hunk 4]
```diff
@@ -133,4 +136,81 @@ contract("pToken", async accounts => {
...
+        amount,
+        { from: governance }
+      );
+      await targetpToken.deposit(amount, { from: governance})
+
+      expect(await targetpToken.balance({from:governance})).to.be.bignumber.equal(amount);
+    });
+
+    /*
+        Below not working - unsure why
```
Comment (corbinpage) [inline tests/pToken.js:181]: @stephenbyrne99 is this still not working?

File: contracts/protektCore/pToken.sol [hunk 5]
```diff
@@ -53,6 +57,9 @@ contract pToken is
     }
 
     function deposit(uint256 _amount) public {
+        if(!claimsManager.isReady()){
```
Comment (corbinpage) [inline contracts/protektCore/pToken.sol:60]: @stephenbyrne99 probably add a require() statement here instead. Like `require(claimsManager.isReady(), "!Ready");`

File: contracts/claimsManagers/ClaimsManagerSingleAccount.sol (hunk 1 above)
Comment (stephenbyrne99) [reply contracts/claimsManagers/ClaimsManagerSingleAccount.sol:28]: This is not idented locally - not sure why it is on the PR

File: tests/pToken.js (hunk 4 above)
Comment (stephenbyrne99) [reply tests/pToken.js:181]: nope

File: contracts/claimsManagers/ClaimsManagerSingleAccount.sol (hunk 1 above)
Comment (corbinpage) [reply contracts/claimsManagers/ClaimsManagerSingleAccount.sol:28]: Looks like it might be using tabs instead of spaces, causing the weird indention.

**Pull request merged**